pkg/cloud/instance.go

```diff
@@ -102,14 +102,14 @@ func (c *client) ResolveServiceOffering(csMachine *infrav1.CloudStackMachine) (o
 		}
 		return csMachine.Spec.Offering.ID, nil
 	}
-	offeringID, count, err := c.cs.ServiceOffering.GetServiceOfferingID(csMachine.Spec.Offering.Name)
+	offeringID, count, err := c.cs.ServiceOffering.GetServiceOfferingID(csMachine.Spec.Offering.Name, cloudstack.WithZone(zoneID))
```

Comment:
Nice on using withZone function instead of listing service offerings!

Reply:
it uses list service offering internally, just seteting zoneID as additional parameter before calling.